sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalog.scala

```diff
@@ -139,7 +139,10 @@ class JDBCTableCatalog extends TableCatalog with Logging {
     checkNamespace(ident.namespace())
     withConnection { conn =>
       classifyException(s"Failed table altering: $ident") {
-        JdbcUtils.alterTable(conn, getTableName(ident), changes, options)
+        val optionsWithTableName = new JDBCOptions(
+          options.parameters + (JDBCOptions.JDBC_TABLE_NAME -> getTableName(ident)))
+        val tableSchema: StructType = JdbcUtils.getSchemaOption(conn, optionsWithTableName).get
```

Comment:
ah now I get what you mean. The `AlterTable` logical plan does have the table schema, but the catalog API doesn't pass it in. It's not possible to change the catalog API at this point, and it's also not worthy to add an extra table lookup here just to support update nullability in MySQL.

I think the first version is fine. Sorry for the back and forth!

Reply:
Thanks will do so ! 👍